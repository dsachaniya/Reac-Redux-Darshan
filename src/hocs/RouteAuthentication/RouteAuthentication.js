/* eslint-disable react/prop-types */
import React from 'react';
import { Route, Redirect } from 'react-router-dom';

/**
 *
 * @param {Node} component
 * @param {boolean} isLoggedIn
 * @param {Object} rest
 * @returns {Node}
 */
function RouteAuthentication({ component: Component, isLoggedIn, ...rest }) {
  return (
    <Route
      {...rest}
      render={(props) =>
        isLoggedIn ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: props.location },
            }}
          />
        )
      }
    />
  );
}

export default RouteAuthentication;
