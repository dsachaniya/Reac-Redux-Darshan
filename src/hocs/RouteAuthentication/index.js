import RouteAuthentication from './RouteAuthentication';

export default RouteAuthentication;
