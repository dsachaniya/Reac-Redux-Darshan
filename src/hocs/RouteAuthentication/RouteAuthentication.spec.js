import React from 'react';
import { shallow } from 'enzyme';
import RouteAuthentication from './RouteAuthentication';

/**
 * TestComponent
 * @returns {Node}
 */
function TestComponent() {
  return <div>TEST</div>;
}

describe('<RouteAuthentication />', () => {
  let wrapper;
  let Component;
  let ComponentWrapper;
  const props = {
    component: TestComponent,
    path: '/test',
    exact: true,
    isLoggedIn: true,
  };

  beforeEach(() => {
    wrapper = shallow(<RouteAuthentication {...props} />);
    Component = wrapper.find('Route').prop('render');
    ComponentWrapper = shallow(<Component />).dive();
  });

  it('rendering', () => {
    const RouteComponent = wrapper.find('Route');
    expect(wrapper).toMatchSnapshot();
    expect(RouteComponent).toHaveLength(1);
    expect(RouteComponent.prop('path')).toEqual('/test');
    expect(RouteComponent.prop('exact')).toEqual(true);
  });

  it('Authenticated Route', () => {
    expect(ComponentWrapper).toMatchSnapshot();
    expect(ComponentWrapper.find('div')).toHaveLength(1);
    expect(ComponentWrapper.find('div').text()).toEqual('TEST');
  });

  it('UnAuthenticated Route', () => {
    wrapper.setProps({ isLoggedIn: false });
    Component = wrapper.find('Route').prop('render');
    ComponentWrapper = shallow(<Component />);
    expect(ComponentWrapper).toMatchSnapshot();
    expect(ComponentWrapper.find('Redirect')).toHaveLength(1);
  });
});
