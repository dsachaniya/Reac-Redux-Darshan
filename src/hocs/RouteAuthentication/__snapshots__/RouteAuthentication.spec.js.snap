// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteAuthentication /> Authenticated Route 1`] = `
<div>
  TEST
</div>
`;

exports[`<RouteAuthentication /> UnAuthenticated Route 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/login",
      "state": Object {
        "from": undefined,
      },
    }
  }
/>
`;

exports[`<RouteAuthentication /> rendering 1`] = `
<Route
  exact={true}
  path="/test"
  render={[Function]}
/>
`;
